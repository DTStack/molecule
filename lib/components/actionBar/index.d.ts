import * as React from 'react';
export interface IActionBarItem<T = any> {
    id: string;
    name?: string;
    title?: string;
    iconName?: string;
    disabled?: boolean;
    data?: T;
    className?: string;
    onClick?(event: React.MouseEvent, item: IActionBarItem): void;
}
export interface IActionBar<T = any> {
    data: IActionBarItem<T>[];
    className?: string;
    onClick?(event: React.MouseEvent, item: IActionBarItem): void;
}
export declare function ActionBarItem(props: IActionBarItem): JSX.Element;
export default function ActionBar<T = any>(props: IActionBar<T>): JSX.Element;
