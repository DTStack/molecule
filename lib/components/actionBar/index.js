"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionBarItem = void 0;
var React = require("react");
var className_1 = require("mo/common/className");
var defaultActionBarClassName = className_1.prefixClaName('action-bar');
var containerClassName = className_1.getBEMElement(defaultActionBarClassName, 'container');
var itemClassName = className_1.getBEMElement(defaultActionBarClassName, 'item');
var itemDisabledClassName = className_1.getBEMModifier(itemClassName, 'disabled');
var labelClassName = className_1.getBEMElement(defaultActionBarClassName, 'label');
function ActionBarItem(props) {
    var id = props.id, title = props.title, name = props.name, onClick = props.onClick;
    var click = function (e) {
        if (onClick) {
            onClick(e, props);
        }
    };
    var disabled = props.disabled ? itemDisabledClassName : null;
    var claNames = className_1.classNames(labelClassName, 'codicon', props.iconName, disabled);
    return (React.createElement("li", { className: className_1.classNames(itemClassName, disabled), onClick: click, key: "" + id },
        React.createElement("a", { className: claNames, title: title }, name)));
}
exports.ActionBarItem = ActionBarItem;
function ActionBar(props) {
    var _a = props.data, data = _a === void 0 ? [] : _a, onClick = props.onClick, className = props.className, custom = __rest(props, ["data", "onClick", "className"]);
    var claNames = className_1.classNames(defaultActionBarClassName, className);
    var items = data.map(function (item) { return (React.createElement(ActionBarItem, __assign({ key: item.id, onClick: onClick }, item))); });
    return (React.createElement("div", __assign({ className: claNames }, custom),
        React.createElement("ul", { className: containerClassName }, items)));
}
exports.default = ActionBar;
//# sourceMappingURL=index.js.map