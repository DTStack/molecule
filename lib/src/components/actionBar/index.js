"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionBarItem = void 0;
var React = require("react");
var react_1 = require("react");
var className_1 = require("mo/common/className");
var contextMenu_1 = require("mo/components/contextMenu");
var dom_1 = require("mo/common/dom");
var menu_1 = require("mo/components/menu");
var utils_1 = require("mo/common/utils");
var defaultActionBarClassName = className_1.prefixClaName('action-bar');
var containerClassName = className_1.getBEMElement(defaultActionBarClassName, 'container');
var itemClassName = className_1.getBEMElement(defaultActionBarClassName, 'item');
var itemDisabledClassName = className_1.getBEMModifier(itemClassName, 'disabled');
var itemCheckedClassName = className_1.getBEMModifier(itemClassName, 'checked');
var labelClassName = className_1.getBEMElement(defaultActionBarClassName, 'label');
function ActionBarItem(props) {
    var id = props.id, title = props.title, name = props.name, _a = props.data, data = _a === void 0 ? {} : _a, _b = props.contextMenu, contextMenu = _b === void 0 ? [] : _b, onClick = props.onClick, onContextMenuClick = props.onContextMenuClick;
    var disabled = props.disabled ? itemDisabledClassName : null;
    var checked = props.checked ? itemCheckedClassName : null;
    var claNames = className_1.classNames(labelClassName, 'codicon', props.iconName, disabled, checked);
    var contextViewMenu;
    var onClickMenuItem = react_1.useCallback(function (e, item) {
        onContextMenuClick === null || onContextMenuClick === void 0 ? void 0 : onContextMenuClick(e, item);
        contextViewMenu === null || contextViewMenu === void 0 ? void 0 : contextViewMenu.dispose();
    }, [contextMenu]);
    var renderContextMenu = function () { return (React.createElement(menu_1.Menu, { onClick: onClickMenuItem, data: contextMenu })); };
    react_1.useEffect(function () {
        if (contextMenu.length > 0) {
            contextViewMenu = contextMenu_1.useContextMenu({
                anchor: dom_1.select("#" + id),
                render: renderContextMenu,
            });
        }
        return function cleanup() {
            contextViewMenu === null || contextViewMenu === void 0 ? void 0 : contextViewMenu.dispose();
        };
    });
    var onClickItem = function (event) {
        if (onClick) {
            onClick(event, props);
        }
        if (contextMenu.length > -1 && contextViewMenu) {
            contextViewMenu.show({
                x: event.clientX,
                y: event.clientY,
            });
        }
    };
    return (React.createElement("li", { id: id, className: className_1.classNames(itemClassName, disabled), onClick: onClickItem, "data-id": data.id },
        React.createElement("a", { className: claNames, title: title }, name)));
}
exports.ActionBarItem = ActionBarItem;
function ActionBar(props) {
    var _a = props.data, data = _a === void 0 ? [] : _a, onClick = props.onClick, onContextMenuClick = props.onContextMenuClick, className = props.className, custom = __rest(props, ["data", "onClick", "onContextMenuClick", "className"]);
    var claNames = className_1.classNames(defaultActionBarClassName, className);
    var items = data.map(function (item, index) { return (React.createElement(ActionBarItem, __assign({ key: item.id }, item, { onContextMenuClick: onContextMenuClick, "data-index": index, onClick: utils_1.mergeFunctions(onClick, item.onClick) }))); });
    return (React.createElement("div", __assign({ className: claNames }, custom),
        React.createElement("ul", { className: containerClassName }, items)));
}
exports.default = ActionBar;
//# sourceMappingURL=index.js.map