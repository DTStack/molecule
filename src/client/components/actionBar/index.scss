@import '../../classNames/style.scss';

$actionBar: prefix('actionBar');
$container: BEMElement($actionBar, 'container');

.#{$container} {
    color: var(--foreground);
    height: 100%;
    align-items: center;
    display: flex;
    padding: 0;
    font-size: 0;
    line-height: 0;

    &:empty {
        width: 0;
        height: 0;
    }
}

:export {
    container: $container;
}