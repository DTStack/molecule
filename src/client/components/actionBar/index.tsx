import React from 'react';
import { classNames } from 'mo/client/classNames';
import { useConnector, useLocale } from 'mo/client/hooks';
import type { HTMLElementProps, IMenuItemProps } from 'mo/types';
import { classify, getNameForTitle, sortByIndex } from 'mo/utils';

import Action from '../action';
import Dropdown from '../dropdown';
import Prevent from '../prevent';
import variables from './index.scss';

export interface IActionBarProps extends HTMLElementProps {
    data?: IMenuItemProps[];
    onClick?: (item: IMenuItemProps) => void;
}

export default function ActionBar({ data, style, className, title, role, onClick }: IActionBarProps) {
    const localize = useLocale();
    const builtin = useConnector('builtin');
    if (!data) return null;
    const [inline = [], ellipsis = []] = classify(data, (i) => i.group === 'inline');
    return (
        <Prevent className={classNames(variables.container, className)} style={style} title={title} role={role}>
            {inline.sort(sortByIndex).map((i) => (
                <React.Fragment key={i.id}>
                    {i.render?.(i) || (
                        <Action
                            type={i.icon}
                            title={i.title || getNameForTitle(i.name)}
                            disabled={i.disabled}
                            onClick={() => onClick?.(i)}
                        >
                            {i.name}
                        </Action>
                    )}
                </React.Fragment>
            ))}
            {!!ellipsis.length && (
                <Dropdown data={ellipsis} trigger="click" onClick={onClick} stopPropagation>
                    <Action
                        type="ellipsis"
                        title={localize(builtin.constants.SIDEBAR_TOOLBAR_ELLIPSIS, 'View and More Actions...')}
                    />
                </Dropdown>
            )}
        </Prevent>
    );
}
