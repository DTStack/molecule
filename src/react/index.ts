export * from './component';
export * from './helper';
export * from './connector';
export * from './controller';
