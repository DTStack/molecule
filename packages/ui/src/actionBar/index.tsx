import React from 'react';
import { useCallback, useEffect, useRef } from 'react';
import {
    prefixClaName,
    classNames,
    getBEMElement,
    getBEMModifier,
    mergeFunctions,
    getDataAttributionsFromProps,
} from '@dtinsight/molecule-common';
import type { HTMLElementProps, UniqueId } from '@dtinsight/molecule-common';
import Tooltip from '../tooltip';
import { Icon } from '../icon';
import { IContextMenu, useContextMenu } from '../contextMenu';
import { IMenuItemProps, Menu } from '../menu';

export interface IActionBarItemProps<T = any> {
    id: UniqueId;
    title?: string | JSX.Element;
    name?: React.ReactNode;
    icon?: string | JSX.Element;
    disabled?: boolean;
    checked?: boolean;
    data?: T;
    contextMenu?: IMenuItemProps[];
    onContextMenuClick?: (
        e: React.MouseEvent,
        item: IMenuItemProps | undefined
    ) => void;
    onClick?(event: React.MouseEvent, item: IActionBarItemProps): void;

    [key: string]: any;
}

export interface IActionBarProps<T = any> extends HTMLElementProps {
    data: IActionBarItemProps<T>[];
    onContextMenuClick?: (
        e: React.MouseEvent,
        item: IMenuItemProps | undefined
    ) => void;
    onClick?(event: React.MouseEvent, item: IActionBarItemProps): void;

    [key: string]: any;
}

const defaultActionBarClassName = prefixClaName('action-bar');
const containerClassName = getBEMElement(
    defaultActionBarClassName,
    'container'
);
const itemClassName = getBEMElement(defaultActionBarClassName, 'item');
const itemDisabledClassName = getBEMModifier(itemClassName, 'disabled');
const itemCheckedClassName = getBEMModifier(itemClassName, 'checked');

export function ActionBarItem(props: IActionBarItemProps) {
    const {
        title,
        name,
        contextMenu = [],
        onClick,
        icon,
        onContextMenuClick,
        ...restProps
    } = props;
    const disabled = props.disabled ? itemDisabledClassName : null;
    const checked = props.checked ? itemCheckedClassName : null;
    const refItem = useRef<HTMLLIElement>(null);

    const contextViewMenu = useRef<IContextMenu>();

    const onClickMenuItem = useCallback(
        (e: React.MouseEvent, item: IMenuItemProps | undefined) => {
            onContextMenuClick?.(e, item);
            contextViewMenu.current?.hide();
        },
        [contextMenu]
    );
    const renderContextMenu = () => (
        <Menu onClick={onClickMenuItem} data={contextMenu} />
    );

    useEffect(() => {
        if (contextMenu.length > 0) {
            contextViewMenu.current = useContextMenu({
                anchor: refItem.current,
                render: renderContextMenu,
            });
        }
        return function cleanup() {
            contextViewMenu.current?.dispose();
        };
    });

    const onClickItem = function (event) {
        if (props.disabled) {
            event.preventDefault();
            return;
        }
        if (onClick) {
            onClick(event, props);
        }
        if (contextMenu.length > -1 && contextViewMenu.current) {
            contextViewMenu.current.show({
                x: event.clientX,
                y: event.clientY,
            });
        }
    };

    const dataProps = getDataAttributionsFromProps(restProps);

    return (
        <li
            ref={refItem}
            className={classNames(itemClassName, disabled, checked)}
            onClick={onClickItem}
            {...dataProps}
        >
            <Tooltip overlay={<span>{title}</span>}>
                <Icon type={icon}>{name}</Icon>
            </Tooltip>
        </li>
    );
}

export function ActionBar<T = any>(props: IActionBarProps<T>) {
    const {
        data = [],
        onClick,
        onContextMenuClick,
        className,
        style,
        title,
        ...restProps
    } = props;

    const claNames = classNames(defaultActionBarClassName, className);

    const items = data.map((item, index) => (
        <ActionBarItem
            key={item.id}
            {...item}
            onContextMenuClick={onContextMenuClick}
            data-index={index}
            onClick={mergeFunctions(onClick, item.onClick)}
        />
    ));

    const dataProps = getDataAttributionsFromProps(restProps);

    return (
        <div className={claNames} style={style} title={title} {...dataProps}>
            <ul className={containerClassName}>{items}</ul>
        </div>
    );
}
