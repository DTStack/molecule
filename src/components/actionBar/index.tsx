import React from 'react';
import { useCallback, useEffect, useRef } from 'react';
import {
    prefixClaName,
    classNames,
    getBEMElement,
    getBEMModifier,
} from 'mo/common/className';
import { useContextMenu } from 'mo/components/contextMenu';
import { IMenuItemProps, Menu } from 'mo/components/menu';
import { mergeFunctions } from 'mo/common/utils';
import Tooltip from '../tooltip';
import { Icon } from '../icon';

export interface IActionBarItemProps<T = any> {
    id?: string;
    name?: React.ReactNode;
    title?: string;
    icon?: string | JSX.Element;
    disabled?: boolean;
    checked?: boolean;
    data?: T;
    contextMenu?: IMenuItemProps[];
    className?: string;
    onContextMenuClick?: (
        e: React.MouseEvent,
        item: IMenuItemProps | undefined
    ) => void;
    onClick?(event: React.MouseEvent, item: IActionBarItemProps): void;
}

export interface IActionBarProps<T = any> {
    data: IActionBarItemProps<T>[];
    className?: string;
    onContextMenuClick?: (
        e: React.MouseEvent,
        item: IMenuItemProps | undefined
    ) => void;
    onClick?(event: React.MouseEvent, item: IActionBarItemProps): void;
}

const defaultActionBarClassName = prefixClaName('action-bar');
const containerClassName = getBEMElement(
    defaultActionBarClassName,
    'container'
);
const itemClassName = getBEMElement(defaultActionBarClassName, 'item');
const itemDisabledClassName = getBEMModifier(itemClassName, 'disabled');
const itemCheckedClassName = getBEMModifier(itemClassName, 'checked');

export function ActionBarItem(props: IActionBarItemProps) {
    const {
        id,
        title,
        name,
        data = {},
        contextMenu = [],
        onClick,
        icon,
        onContextMenuClick,
    } = props;
    const disabled = props.disabled ? itemDisabledClassName : null;
    const checked = props.checked ? itemCheckedClassName : null;
    const refItem = useRef(null);

    let contextViewMenu;

    const onClickMenuItem = useCallback(
        (e: React.MouseEvent, item: IMenuItemProps | undefined) => {
            onContextMenuClick?.(e, item);
            contextViewMenu?.dispose();
        },
        [contextMenu]
    );
    const renderContextMenu = () => (
        <Menu onClick={onClickMenuItem} data={contextMenu} />
    );

    useEffect(() => {
        if (contextMenu.length > 0) {
            contextViewMenu = useContextMenu({
                anchor: refItem.current,
                render: renderContextMenu,
            });
        }
        return function cleanup() {
            contextViewMenu?.dispose();
        };
    });

    const onClickItem = function (event) {
        if (props.disabled) {
            event.preventDefault();
            return;
        }
        if (onClick) {
            onClick(event, props);
        }
        if (contextMenu.length > -1 && contextViewMenu) {
            contextViewMenu.show({
                x: event.clientX,
                y: event.clientY,
            });
        }
    };

    return (
        <li
            id={id}
            ref={refItem}
            className={classNames(itemClassName, disabled, checked)}
            onClick={onClickItem}
            data-id={data.id}
        >
            <Tooltip overlay={<span>{title}</span>}>
                <Icon type={icon}>{name}</Icon>
            </Tooltip>
        </li>
    );
}

export function ActionBar<T = any>(props: IActionBarProps<T>) {
    const {
        data = [],
        onClick,
        onContextMenuClick,
        className,
        ...custom
    } = props;

    const claNames = classNames(defaultActionBarClassName, className);

    const items = data.map((item: IActionBarItemProps<T>, index) => (
        <ActionBarItem
            key={item.id}
            {...item}
            onContextMenuClick={onContextMenuClick}
            data-index={index}
            onClick={mergeFunctions(onClick, item.onClick)}
        />
    ));

    return (
        <div className={claNames} {...custom}>
            <ul className={containerClassName}>{items}</ul>
        </div>
    );
}
