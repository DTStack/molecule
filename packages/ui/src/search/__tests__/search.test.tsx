import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { IActionBarItemProps } from '../../actionBar';
import Input, { InfoTypeEnums } from '../input';
import { Search } from '../index';
import {
    replaceBtnClassName,
    replaceContainerClassName,
    validationInfoInputClassName,
    validationWarningInputClassName,
    validationBaseInputClassName,
} from '../base';

const { Group } = Input;

const TEST_ID = 'test';

const mockData: IActionBarItemProps[] = [
    {
        id: TEST_ID,
        name: 'mockDataTitle',
        title: 'mockDataTitle',
        icon: 'add',
        checked: true,
        'data-testid': TEST_ID,
    },
];

describe('Test Input Component', () => {
    test('Match the Group Input snapshot', () => {
        const componentInput = renderer.create(<Input />);
        const treeInput = componentInput.toJSON();

        expect(treeInput).toMatchSnapshot();

        const componentGroupWithChildren = renderer.create(
            <Group>
                <Input />
                <Input />
            </Group>
        );
        const treeChilren = componentGroupWithChildren.toJSON();

        expect(treeChilren).toMatchSnapshot();
    });

    test('Input with Group snapshot', () => {
        render(
            <Group>
                <Input />
            </Group>
        );

        const textArea = screen.getByText(/.*/i, { selector: 'textarea' });

        expect(textArea).not.toBeNull();
    });

    test('Input className', () => {
        const INPUT_CLASSNAME = 'test-name';

        render(<Input className={INPUT_CLASSNAME} />);

        const div = screen.getByText(/.*/i, {
            selector: 'textarea',
        }).parentNode as HTMLDivElement;

        expect(div.classList).toContain(INPUT_CLASSNAME);
    });

    test('Input placeholder', () => {
        const INPUT_PLACEHOLDER = 'molecule';

        render(<Input placeholder={INPUT_PLACEHOLDER} />);

        const div = screen.getByText(/.*/i, {
            selector: 'textarea',
        });

        expect(div.getAttribute('placeholder')).toBe(INPUT_PLACEHOLDER);
    });

    test('Input info', () => {
        const INPUT_INFO = { type: InfoTypeEnums.warning, text: TEST_ID };
        render(<Input info={INPUT_INFO} />);

        const textArea = screen.getByText(/.*/i, {
            selector: 'textarea',
        });
        expect(textArea.classList).toContain(validationWarningInputClassName);

        fireEvent.focus(textArea);
        const div = textArea.nextElementSibling as HTMLDivElement;

        expect(div.textContent).toBe(TEST_ID);
        expect(div.classList).toContain(validationBaseInputClassName);
        expect(div.classList).toContain(validationWarningInputClassName);
    });

    test('Input change', () => {
        const TEST_FN = jest.fn();
        const wrapper = render(<Input onChange={TEST_FN} />);
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );

        fireEvent.keyDown(textarea!, {
            key: 'Enter',
            keyCode: 13,
        });
        expect(TEST_FN).toBeCalled();
    });
});

describe('Test Search Component', () => {
    test('Match the Search snapshot', () => {
        const componentInput = renderer.create(<Search />);
        const search = componentInput.toJSON();

        expect(search).toMatchSnapshot();
    });

    test('Search className', () => {
        const INPUT_CLASSNAME = 'test-name';
        const wrapper = render(<Search className={INPUT_CLASSNAME} />);
        const div = wrapper.container.firstElementChild as HTMLDivElement;

        expect(div.classList).toContain(INPUT_CLASSNAME);
    });

    test('Search style', () => {
        const INPUT_STYLE = { background: 'red' };
        const wrapper = render(<Search style={INPUT_STYLE} />);
        const div = wrapper.container.firstElementChild as HTMLDivElement;

        expect(div.getAttribute('style')).toBe('background: red;');
    });

    test('Search value', () => {
        const INPUT_VALUE = 'test';
        const wrapper = render(
            <Search data-testid={TEST_ID} values={[INPUT_VALUE]} />
        );
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );

        expect(textarea!.value).toBe(INPUT_VALUE);
    });

    test('Search dafult placeholder', () => {
        const INPUT_VALUE = 'test';
        const wrapper = render(<Search values={[INPUT_VALUE]} />);
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );

        expect(textarea!.value).toBe(INPUT_VALUE);
    });

    test('Search placeholder', () => {
        const DEFAULT_VALUE = 'Search';
        const wrapper = render(<Search />);
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );

        expect(textarea!.placeholder).toBe(DEFAULT_VALUE);
    });

    test('Search addons', () => {
        const { getByTestId } = render(<Search addons={[mockData]} />);

        expect(getByTestId(TEST_ID)).toBeInTheDocument();
    });

    test('Search validationInfo', () => {
        const TEST_TEXT = 'molecule';
        const wrapper = render(<Search validationInfo={TEST_TEXT} />);
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );
        expect(textarea!.classList).toContain(validationInfoInputClassName);

        fireEvent.focus(textarea!);
        const div = textarea!.nextElementSibling as HTMLDivElement;
        expect(div!.classList).toContain(validationInfoInputClassName);
        expect(div!.textContent).toContain(TEST_TEXT);
    });

    test('Search change event', () => {
        const TEST_CHANGE_FN = jest.fn();
        const TEST_SEARCH_FN = jest.fn();
        const TEST_VALUE = 'molecule';
        const wrapper = render(
            <Search onChange={TEST_CHANGE_FN} onSearch={TEST_SEARCH_FN} />
        );
        const textarea = wrapper.container.querySelector<HTMLTextAreaElement>(
            `textarea[autoCorrect='off']`
        );

        fireEvent.change(textarea!, { target: { value: TEST_VALUE } });
        expect(TEST_CHANGE_FN).toBeCalled();
        expect(TEST_SEARCH_FN).toBeCalled();
    });

    test('Search onButtonClick and replace show', () => {
        const TEST_FN = jest.fn();
        const INPUT_VALUE = 'test';
        const wrapper = render(
            <Search
                addons={[, mockData]}
                values={[, INPUT_VALUE]}
                onButtonClick={TEST_FN}
            />
        );
        const icon =
            wrapper.container.getElementsByClassName(replaceBtnClassName)[0];
        expect(icon).not.toBeNull();

        fireEvent.click(icon);
        expect(TEST_FN).toBeCalled();

        const input = wrapper.container.getElementsByClassName(
            replaceContainerClassName
        )[0];
        expect(input).toBeInTheDocument();

        const textarea =
            wrapper.container.querySelectorAll<HTMLTextAreaElement>(
                `textarea[autoCorrect='off']`
            )[1];
        expect(textarea!.value).toBe(INPUT_VALUE);

        expect(wrapper.getByTestId(TEST_ID)).toBeInTheDocument();
    });
});
