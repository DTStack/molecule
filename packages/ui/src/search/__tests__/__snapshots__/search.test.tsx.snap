// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component Match the Group Input snapshot 1`] = `
<div>
  <textarea
    autoCapitalize="off"
    autoCorrect="off"
    className=""
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    spellCheck={false}
    value=""
  />
  <div
    className="mo-action-bar mo-search__toolbar"
  >
    <ul
      className="mo-action-bar__container"
    />
  </div>
</div>
`;

exports[`Test Input Component Match the Group Input snapshot 2`] = `
<div
  className="mo-search__group"
>
  <div>
    <textarea
      autoCapitalize="off"
      autoCorrect="off"
      className=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      spellCheck={false}
      value=""
    />
    <div
      className="mo-action-bar mo-search__toolbar"
    >
      <ul
        className="mo-action-bar__container"
      />
    </div>
  </div>
  <div>
    <textarea
      autoCapitalize="off"
      autoCorrect="off"
      className=""
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      spellCheck={false}
      value=""
    />
    <div
      className="mo-action-bar mo-search__toolbar"
    >
      <ul
        className="mo-action-bar__container"
      />
    </div>
  </div>
</div>
`;

exports[`Test Search Component Match the Search snapshot 1`] = `
<div
  className="mo-search"
>
  <span
    className="mo-search__replace__button codicon codicon-chevron-right"
    onClick={[Function]}
  />
  <div
    className="mo-search__group"
  >
    <div
      className="mo-search__input mo-search__target"
    >
      <textarea
        autoCapitalize="off"
        autoCorrect="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search"
        spellCheck={false}
        title="Search"
        value=""
      />
      <div
        className="mo-action-bar mo-search__toolbar"
      >
        <ul
          className="mo-action-bar__container"
        />
      </div>
    </div>
  </div>
</div>
`;
